lib/apiservers/service/restapi/handlers/common.go

```diff
@@ -186,18 +187,32 @@ func getVCHConfig(op trace.Operation, d *data.Data, validator *validate.Validato
 	return vchConfig, nil
 }
 
-func getAddresses(vchConfig *config.VirtualContainerHostConfigSpec) (dockerHost, adminPortal string) {
-	if client := vchConfig.ExecutorConfig.Networks["client"]; client != nil {
-		if publicIP := client.Assigned.IP; publicIP != nil {
-			var dockerPort = opts.DefaultTLSHTTPPort
-			if vchConfig.HostCertificate.IsNil() {
-				dockerPort = opts.DefaultHTTPPort
-			}
+func getAddresses(executor *management.Dispatcher, vchConfig *config.VirtualContainerHostConfigSpec) (string, string, error) {
+	clientNet := vchConfig.ExecutorConfig.Networks["client"]
+	if clientNet != nil {
+		clientIP := clientNet.Assigned.IP
+
+		if ip.IsUnspecifiedIP(clientIP) {
+			return "", "", fmt.Errorf("No client IP address assigned")
+		}
 
-			dockerHost = fmt.Sprintf("%s:%d", publicIP, dockerPort)
-			adminPortal = fmt.Sprintf("https://%s:%d", publicIP, constants.VchAdminPortalPort)
+		hostIP := clientIP.String()
+		// try looking up preferred name irrespective of CAs, if available
+		// once we found a preferred address, we use that instead of the assigned client IP address
+		if cert, err := vchConfig.HostCertificate.X509Certificate(); err == nil {
+			hostIP = executor.GetTLSFriendlyHostIP(clientIP, cert, vchConfig.CertificateAuthorities)
 		}
+
+		dockerHost := ""
+		if !vchConfig.HostCertificate.IsNil() {
+			dockerHost = fmt.Sprintf("%s:%d", hostIP, opts.DefaultTLSHTTPPort)
```

Comment:
I think the pattern from the old code, where we set a `dockerPort` based on `vchConfig.HostCertificate.IsNil()` and then have a single `dockerHost = fmt.Sprintf("%s:%d", publicIP, dockerPort)` is a little cleaner.

Reply:
👍 fixed